### PR TITLE
Test failbit when converting boundary ids

### LINKED ISSUE #24950: Check for failure to convert `boundary_id_type`
## Reason
For efficiency we still default `boundary_id_type` to 16-bit (signed, for some reason...), so users can easily manually specify a `BoundaryName` which is an integer (so we try to convert it to an id directly rather than treating it as a string name) but which fails to fit in `boundary_id_type` (so conversion via `istream` just gives us 2^15-1) ... and when this happens we don't test for failure afterward.

## Design
Check the `failbit` on that `istream`, scream an informative error message and die if it's set.

## Impact
It's possible for a user's input file to be fine with the current behavior, which would produce e.g. a boundary name of "40000" attached to a boundary id of 32767, which doesn't *inherently* break anything.  But I suspect any users who have so narrowly avoided being bitten by this bug will be fine changing their files to accommodate its fix.

### PR DESCRIPTION
**Pull request by roystgnr**
Resolves #24950

Pinging @olinwc to tell me if the message is reasonable; this seems to work well on one of his test cases.

### CODE REVIEW

**Review by lindsayad — COMMENTED** [review]
wondering if we can unit test this

**Comment by roystgnr** [comment]
> wondering if we can unit test this

Yeah, but I got lazy and just put in a regression test instead.

File: test/tests/meshgenerators/polyline_mesh_generator/tests [hunk 1]
```diff
@@ -37,4 +37,21 @@
       detail = "refining a loop if requested"
     []
   []
+  [polyline_good_boundary]
+    type = 'Exodiff'
+    input = 'polyline_mesh_generator_good_boundary.i'
+    cli_args = '--mesh-only'
+    exodiff = 'polyline_mesh_generator_good_boundary_in.e'
+    recover = false
+    requirement = "The system shall set a boundary name or id if requested"
```
Comment (lindsayad) [inline test/tests/meshgenerators/polyline_mesh_generator/tests:46]:
```suggestion
    requirement = "The system shall set a boundary name or id if requested."
```

File: test/tests/meshgenerators/polyline_mesh_generator/tests [hunk 2]
```diff
@@ -37,4 +37,21 @@
...
+  []
+  [polyline_bad_boundary]
+    type = 'RunException'
+    input = 'polyline_mesh_generator_bad_boundary.i'
+    expect_err = "Failed to convert integer"
+    issues = '#24950'
+    cli_args = '--mesh-only'
+    recover = false
+    requirement = "The system shall detect any unsatisfiable boundary id requests"
```
Comment (lindsayad) [inline test/tests/meshgenerators/polyline_mesh_generator/tests:55]:
```suggestion
    requirement = "The system shall detect any unsatisfiable boundary id requests."
```

**Review by lindsayad — DISMISSED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**